driver/netconf/capabilities.go

```diff
@@ -47,6 +53,18 @@ func (d *Driver) processServerCapabilities() error {
 		d.serverCapabilities = append(d.serverCapabilities, string(match[1]))
 	}
 
+	// extract session id if it exists in the hello message
+	sessionIDMatch := ncPatterns.sessionID.FindSubmatch(b)
+	if len(sessionIDMatch) != 2 {
+		return nil
+	}
+
+	i, err := strconv.Atoi(string(sessionIDMatch[1]))
+	if err != nil {
+		return fmt.Errorf("invalid sessionID=%s: %v", string(sessionIDMatch[1]), err)
```

Comment:
can we use the `ErrNetconfError` base error here (like up on line 45 or so) I *think* we should be using that most places in here, or at least that was my original plan :D

Reply:
done!